cmd/internal/cli/actions.go

```diff
@@ -244,15 +251,90 @@ var TestCmd = &cobra.Command{
 	Run: func(cmd *cobra.Command, args []string) {
 		a := append([]string{"/.singularity.d/actions/test"}, args[1:]...)
 		setVM(cmd)
-		if VM {
+		if vm {
 			execVM(cmd, args[0], a)
 			return
 		}
-		execStarter(cmd, args[0], a, "")
+		if err := launchContainer(cmd, args[0], a, ""); err != nil {
+			sylog.Fatalf("%s", err)
+		}
 	},
 
 	Use:     docs.RunTestUse,
 	Short:   docs.RunTestShort,
 	Long:    docs.RunTestLong,
 	Example: docs.RunTestExample,
 }
+
+func launchContainer(cmd *cobra.Command, image string, args []string, instanceName string) error {
+	ns := launch.Namespaces{
+		User: userNamespace,
+		UTS:  utsNamespace,
+		PID:  pidNamespace,
+		IPC:  ipcNamespace,
+		Net:  netNamespace,
+	}
+
+	cgJSON, err := getCgroupsJSON()
+	if err != nil {
+		return err
+	}
+
+	ki, err := getEncryptionMaterial(cmd)
+	if err != nil {
+		return err
+	}
+
+	opts := []launch.Option{
```

Comment:
I'm concerned that there will be future programming error cases where people forget to copy the cobra variable to the `launch.Option` structure when adding a new option.  Wouldn't it be possible/better to have cobra write directly into the launch structure?